File: src/main/java/coza/trojanc/stonewriter/template/fields/Text.java
package coza.trojanc.stonewriter.template.fields;

/**
 * A class representing simple static text to be printed
 */
public class Text extends AbstractTextItem{

	/**
	 * Text to be printed
	 */
	private String text;
	/**
	 * Text constructor.
	 */ 
	public Text() {
		super();
	}

	public Text(AbstractTextItem source){
		super(source);
	}


	public String getText() {
		return text;
	}

	public void setText(String text) {
		this.text = text;
	}

	@Override
	public String toString() {
		return new StringBuilder()
			.append("Text[")
			.append("text=").append(this.text)
			.append(",align=").append(this.getAlignment())
			.append(",offset=").append(this.getOffset())
			.append("]")
			.toString();
	}
}

File: receipt-shared/src/main/kotlin/coza/trojanc/receipt/template/fields/Text.kt
package coza.trojanc.receipt.template.fields

/**
 * A class representing simple static text to be printed.
 * @author Charl Thiem
 */
class Text: AbstractTextItem{

    /**
     * Text to be printed
     */
    private var text: String? = null

    /**
     * Text constructor.
     */
    constructor() : super() {
    }

    constructor(source: AbstractTextItem): super(source) {
    }


    fun getText(): String? {
        return text
    }

    fun setText(text: String) {
        this.text = text
    }

    override fun toString(): String {
        return StringBuilder()
                .append("Text[")
                .append("text=").append(this.text)
                .append(",align=").append(this.getAlignment())
                .append(",offset=").append(this.getOffset())
                .append("]")
                .toString()
    }
}
